### PR TITLE
Expose block random seed and timestamp to contract ABI

### PR DESCRIPTION
**Pull request by holygits**
Have found these functions to be useful vs. the complexity of serializing the call through `ext_dispatch_call`

### CODE REVIEW

**Comment by bkchr** [comment]
@holygits please fix the tests.

**Comment by gavofyork** [comment]
This does introduce a new dependency (timestamp). It's fairly reasonable though, I guess.

File: srml/contract/src/tests.rs [hunk 1]
```diff
@@ -78,6 +78,15 @@ impl balances::Trait for Test {
 	type EnsureAccountLiquid = ();
 	type Event = MetaEvent;
 }
+impl timestamp::Trait for Test {
+	type Moment = u64;
+	type OnTimestampSet = ();
+}
+impl consensus::Trait for Test {
```
Comment (pepyakin) [inline srml/contract/src/tests.rs]: Why do we need this one?

File: srml/contract/src/wasm/mod.rs [hunk 2]
```diff
@@ -1027,4 +1035,131 @@ mod tests {
...
+		)
+
+		;; copy contents of the scratch buffer into the contract's memory.
+		(call $ext_scratch_copy
+			(i32.const 32)		;; Pointer in memory to the place where to copy.
+			(i32.const 0)		;; Offset from the start of the scratch buffer.
+			(i32.const 32)		;; Count of bytes to copy.
+		)
+
+		;; assert that contents of the buffer is equal to the i64 value of 0 or H256::default().
```
Comment (pepyakin) [inline srml/contract/src/wasm/mod.rs]: this actually tests 1/4 of the whole value and I'd really prefer test some non-zero value.

File: srml/contract/src/tests.rs (hunk 1 above)
Comment (holygits) [reply srml/contract/src/tests.rs]: timestamp::Trait depends on it to be implemented

**Comment by gavofyork** [comment]
this also needs a bump to the runtime version in `node`.

**Comment by holygits** [comment]
Have bumped spec version

**Comment by gavofyork** [comment]
Needs a merge.

**Pull request merged**